modules/enterprise/gui/portal-war/src/main/java/org/rhq/enterprise/gui/common/paging/PageControlSettingsUIBean.java

```diff
@@ -24,8 +24,25 @@
  * @author Joseph Marques
  */
 public class PageControlSettingsUIBean {
-    private SelectItem[] pageSizes = new SelectItem[] { new SelectItem("15", "15"), new SelectItem("30", "30"),
-        new SelectItem("45", "45") };
+    private SelectItem[] pageSizes;
+    private static String MAX_ITEMS_PER_PAGE = "rhq.server.gui.max-items-per-page";
+
+    public PageControlSettingsUIBean() {
+        try {
+            String maxItemsPerPage = System.getProperty(MAX_ITEMS_PER_PAGE);
+            if (maxItemsPerPage != null && Integer.parseInt(maxItemsPerPage) > 45) {
+                pageSizes = new SelectItem[] { new SelectItem("15", "15"), new SelectItem("30", "30"),
```

Comment:
Uhg, I'm not sure. I might have to test with a ton of items.
I recall seeing in the code a bug when we had lots of pages. Not sure if you are refering to that one. https://github.com/josejulio/rhq/blob/d5dfe2ed9d42af51c6c3d3793f79e315c52ed092/modules/enterprise/gui/portal-war/src/main/webapp/rhq/resource/include/pagination-plain.xhtml#L66-L66